### PR TITLE
JAVA-22208 spring cloud eureka to jdk17

### PR DESCRIPTION
**Pull request by anastasiosioannidis**

### CODE REVIEW

File: spring-cloud-modules/pom.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <modules>
         <module>spring-cloud-loadbalancer</module>
         <module>spring-cloud-config</module>
-        <module>spring-cloud-eureka</module>
+<!--        <module>spring-cloud-eureka</module>-->
```
Comment (dkapil) [inline spring-cloud-modules/pom.xml:22]: mention here that "uncomment when spring-cloud-modules is moved to JDK9+ profile"

File: spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/pom.xml [hunk 2]
```diff
@@ -87,4 +110,9 @@
         </plugins>
     </build>
 
+    <properties>
+        <maven.compiler.source>16</maven.compiler.source>
```
Comment (dkapil) [inline spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/pom.xml:114]: our jenkins now run with JDK 17, can we aim for that?

File: spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/src/test/java/com/baeldung/spring/cloud/client/LoadBalancerBooksClientIntegrationTest.java [hunk 3]
```diff
@@ -62,4 +90,53 @@ public void whenGetBooks_thenTheCorrectBooksShouldBeReturned() {
             new Book("Dune", "Frank Herbert"),
             new Book("Foundation", "Isaac Asimov"))));
     }
+
+    @Test
+    void roundRobbinLoadbalancerWorks() throws IOException {
```
Comment (dkapil) [inline spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/src/test/java/com/baeldung/spring/cloud/client/LoadBalancerBooksClientIntegrationTest.java:95]: typo: Robin...

File: spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/src/test/java/com/baeldung/spring/cloud/client/TestConfig.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+public class TestConfig {
+
+    @Bean(initMethod = "start", destroyMethod = "stop")
+    public WireMockServer mockBooksService() {
+        return new WireMockServer(options().port(80));
+    }
+
+    @Bean(name="secondMockBooksService", initMethod = "start", destroyMethod = "stop")
+    public WireMockServer secondBooksMockService() {
+        return new WireMockServer(options().port(81));
```
Comment (dkapil) [inline spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/src/test/java/com/baeldung/spring/cloud/client/TestConfig.java:22]: will this port be available while running in an integration environment in Jenkins.
I think we can use Spring Boot's LocalServerPort for this or something similar, just check

**Review by dkapil — CHANGES_REQUESTED** [review]

File: spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/src/test/java/com/baeldung/spring/cloud/client/TestConfig.java (hunk 4 above)
Comment (anastasiosioannidis) [reply spring-cloud-modules/spring-cloud-eureka/spring-cloud-eureka-feign-client-integration-test/src/test/java/com/baeldung/spring/cloud/client/TestConfig.java:22]: @dkapil so even though this PR's default and integration JDK9+ checks are passing green, is it possible to fail after merging ?

**Comment by anastasiosioannidis** [comment]
retest this please

**Pull request merged**